celery/backends/base.py

```diff
@@ -647,11 +647,36 @@ def _forget(self, task_id):
 
     def _store_result(self, task_id, result, state,
                       traceback=None, request=None, **kwargs):
+
+        if state in self.READY_STATES:
+            date_done = datetime.datetime.utcnow()
+        else:
+            date_done = None
+
         meta = {
-            'status': state, 'result': result, 'traceback': traceback,
-            'children': self.current_task_children(request),
+            'status': state,
+            'result': result,
+            'traceback': traceback,
             'task_id': bytes_to_str(task_id),
+            'date_done': date_done,
         }
+
+        if request:
+            request_meta = {
+                # Some unit tests mock the properties, casting to string to
```

Comment:
I think this comment refers to the next line (`name` key).

Reply:
moved comment... is this what you intended?

Reply:
It referred to the `name` key, isn't that correct?

Reply:
name, worker, queue.  I think they all break due to funky mocks without a cast.  Do you think I should change it?  If so what do you suggest?  thanks!

Reply:
I see that tests are passing in Travis CI. Could you please clarify here, I am not sure I understand your comment?

Reply:
Sorry, I meant that several tests break on name, worker, queue attributes, because the tests do mocks with weird values (e.g. Mock() instances)  .   For regular usage (not tests) the cast to string is not really needed, as it's a str->str op.  For the mentioned tests, the value ends up as a string repr of the Mock, (or in I think one case, a different class).

Just forcing to a string is draconian but effective in this case.

Reply:
@johnarnold I am skeptical about this type casting, I don't think it is advisable to modify code logic for testing purposes. Could you revert and let Travis fail so I can examine the broken tests? I can then suggest changes on the tests.

Reply:
@georgepsarakis I reverted the type casting.

Reply:
@georgepsarakis I did a couple rounds of test fixups:
- changing from Mock to Context() where possible to avoid json serialization errors
- changing Mock to PicklableMock to avoid pickle serialization errors
- calling app.set_current() where needed to avoid hitting the celery app trap

Unfortunately there are still some failing tests, e.g. in t/unit/worker/test_strategy.py  because they're using Mock() in properties of a Context object, and it's not super easy to fix, afaik.

Open to suggestions on how to fix.  Or, can go back to draconian type casting ?  Thanks!